vault/resource_database_secret_backend_connection.go

```diff
@@ -831,10 +869,14 @@ func getDatabaseAPIDataForEngine(engine *dbEngine, idx int, d *schema.ResourceDa
 		setDatabaseConnectionDataWithDisableEscaping(d, prefix, data)
 	case dbEngineElasticSearch:
 		setElasticsearchDatabaseConnectionData(d, prefix, data)
+	case dbEngineRedisElastiCache:
+		setRedisElastiCacheDatabaseConnectionData(d, prefix, data)
 	case dbEngineSnowflake:
 		setDatabaseConnectionDataWithUserPass(d, prefix, data)
 	case dbEngineRedshift:
 		setDatabaseConnectionDataWithDisableEscaping(d, prefix, data)
+	default:
+		return nil, fmt.Errorf("unrecognized DB engine: %v", engine)
```

Comment:
I added a guard since engines are specific enum values and should never silently pass through. Switch cases should [almost always contain a default case](https://stackoverflow.com/questions/4649423/should-switch-statements-always-contain-a-default-clause).

Reply:
Nice! That's a good call, thanks for adding that